lib/tasks/npm-task.js

```diff
@@ -2,21 +2,25 @@
 
 // Runs `npm install` in cwd
 
+const path = require('path');
 const chalk = require('chalk');
 const execa = require('execa');
 const semver = require('semver');
 const RSVP = require('rsvp');
 const SilentError = require('silent-error');
+const existsSync = require('exists-sync');
 
 const logger = require('heimdalljs-logger')('ember-cli:npm-task');
 
 const Task = require('../models/task');
 
 class NpmTask extends Task {
 
-  constructor(options) {
+  constructor(options = {}) {
     super(options);
 
+    this.useYarn = ('useYarn' in options) ? options.useYarn : undefined;
```

Comment:
Should be able to just do:

```js
this.useYarn = options.useYarn;
```

Basically `options.useYarn` is `undefined` if `useYarn` wasn't included in `options` originally...